python/cudf/cudf/core/column/string.py

```diff
@@ -486,6 +486,18 @@ def __reduce__(self):
     def str(self, index=None):
         return StringMethods(self, index=index)
 
+    def __sizeof__(self):
+        n = self.str().len().sum() * self.dtype.itemsize
```

Comment:
@kkraus14 - Would this be the correct way to get the memory footprint of a string column?

Reply:
@davidwendt can you comment?

Reply:
I don't see how this would be accurate for strings since each string can have different lengths.

Reply:
This is all going to be completely reworked in 0.12 as well with using `cudf::column` for strings instead of `nvstrings`.

Reply:
> I don't see how this would be accurate for strings since each string can have a different length.

@davidwendt - `.len()` is producing a series containing the length of each string, and then `.sum()` is reducing the series into a single sum.  That sum (of all string characters) is then multiplied by the dtype size (8 bytes).  How correct/incorrect is this approach?

Reply:
@rjzamora that is the number of characters in each string, and each character can be anywhere from 1 - 4 bytes depending on the character.

Reply:
> This is all going to be completely reworked in 0.12 as well with using cudf::column for strings instead of nvstrings.

Understood - I'd still like to get something "correct" here merged in the short term,  so that we can do proper "smart repartitioning" in dask.

Reply:
You'd want to use `byte_count` and then sum that to get the memory usage.

@davidwendt That wouldn't account for the extra character metadata, but that should be relatively small, correct?

Reply:
If this is an `nvstrings` instance you can call its `device_memory()` member function to get the total size including the meta-data.

Reply:
Somehow missed that, perfect!

Reply:
Perfect - Thanks @davidwendt !